RELEASING.md

```diff
@@ -0,0 +1,60 @@
+# Release Checklist
+
+These steps assume that you've checked out the Soketto repository and are in the root directory of it.
+
+1. Ensure that everything you'd like to see released is on the `develop` branch.
+
+2. Create a release branch off `develop`, for example `release-v0.6.0`. The branch name should start with `release`
+   in order that we can target commits with CI. You'll want to decide how far the version needs to be bumped 
+   based on the changes to date. If unsure what to bump the version to (eg is it a major, minor or patch 
+   release), check with the Parity Tools team.
+
+3. Check that you're happy with the current documentation.
+   
+   ```
+   cargo doc --open
+   ```
+
+   CI checks for broken internal links at the moment. Optionally you can also confirm that any external links
+   are still valid like so:
+
+   ```
+   cargo install cargo-deadlinks
+   cargo deadlinks --check-http
+   ```
+
+   If there are minor issues with the documentation, they can be fixed in the release branch.
+
+4. Bump the crate version in `Cargo.toml` to whatever was decided in step 2.
+
+5. Update `CHANGELOG.md` to reflect the difference between this release and last. If you're unsure of
+   what to add, check with the Tools team. 
+   
+   One way to gain some inspiration about what to write is by looking through the commit history since 
+   the last version (eg `git log --pretty LAST_VERSION_TAG..HEAD`).
+
+   Alternately, look at the commit history: https://github.com/paritytech/soketto/commits/develop.
+
+6. Commit any of the above changes to the release branch and open a PR in GitHub with a base of `develop`.
+
+7. Once the branch has been reviewed, we are ready to publish this branch to crates.io.
+   
+   First, do a dry run to make sure that things seem sane:
+   ```
+   cargo publish --dry-run
+   ```
+
+   If we're happy with everything, proceed with the release:
+   ```
+   cargo publish
+   ```
+
+8. If the release was successful, then tag the latest commit in our release branch so that somebody can
+   get back to this exact state in the future.
```

Comment:
What we do on other projects is we tag `master` once the release PR is merged. I am actually not sure if that makes a difference, but I think we should be consistent with where we are tagging. Here in `soketto` we have the `develop`/`master` dichotomy so I'm even less sure what is right.

@niklasad1 thoughts?

Reply:
I was originally going to suggest merging to `master` (and we still could), but I guess it felt cleanest to release and tag the exact commit that we're  happy with in the release branch instead of merging that somewhere else first and releasing it.

One of the reasons to use `master` is to have a nice record of release commits, but our master branch is a bit messy and not super helpful for that I feel. Another reason is to run CI automatically to publish/deploy on merge to master, but we don't do that yet either.

Ultimately I'd be happy adding a "merge to master" step at the end if we liked ,but it doesn't serve much purpose it feels like at the moment.

Reply:
(tweaked process to merge to `master` then publish then tag then merge master up to `develop`)